### PR TITLE
LIVY-205. sparkr.zip is unzipped on executor side

### PR DESCRIPTION
**Pull request by zjffdu**
This PR is to unzipped sparkr.zip on executor side.

### CODE REVIEW

**Comment by zjffdu** [comment]
@vanzin Although this PR can make it work, but it needs some configuration change on spark-defaults.conf. Because livy use `R` to launch R script while spark use `Rscript`, is there any reason why livy doesn't use `Rscript` ?

**Comment by zjffdu** [comment]
Never mind, I think I may made a mistake in #177.  R is used in spark as well for interactive mode while RScript is used for batch mode.

**Comment by zjffdu** [comment]
@vanzin  I have a Spark PR (https://github.com/apache/spark/pull/14744) to allow setting sparkr shell command through configuration. And I updated this livy PR based on the spark PR.

**Comment by zjffdu** [comment]
ping @vanzin

File: server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala [hunk 1]
```diff
@@ -292,7 +292,7 @@ class InteractiveSession(
         val path = Seq(sparkHome, "R", "lib", "sparkr.zip").mkString(File.separator)
         val rArchivesFile = new File(path)
         require(rArchivesFile.exists(), "sparkr.zip not found; cannot run sparkr application.")
-        rArchivesFile.getAbsolutePath()
+        rArchivesFile.getAbsolutePath() + "#sparkr"
```
Comment (vanzin) [inline server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala]: Don't you need to add this suffix when the user sets `SPARKR_PACKAGE` too? Avoids having to force the user to do it manually.

Comment (zjffdu) [inline server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala]: Yeah, right.

**Comment by zjffdu** [comment]
ping @vanzin

**Comment by vanzin** [comment]
Merging.

**Pull request closed without merging**